### ISSUE TITLE
&str.splitn() not implemented

### ISSUE THREAD

**Issue by Wazubaba** [rust-lang/rust#24075]
To start with, I'm relatively new to the language, and I'm trying to use `&str.splitn()`. According to what I could determine from the API, it should have an `nth()` member, yet it doesn't seem to. It also seems to be missing other members too.

My test is:
```
    let var = "meow".splitn(1, "e").nth(1);
```
Which gives me the error:
```
    error: type `core::str::SplitN<'_, &str>` does not implement any method in scope named `nth` 
```
This however allows it to work: http://is.gd/Vdjy8i (Thanks to Steveklabnik from the rust IRC channel)

This is using ```rustc 1.0.0-beta (9854143cb 2015-04-02) (built 2015-04-02)```.

**Comment by steveklabnik** [comment]
So, while that doesn't work, passing a closure as a pattern does:

```
let var = "meow".splitn(1, {|x| x == 'm' }).nth(1);
```

So there's something about `SplitN` that's not properly fufilling the `Iterator` implementation

**Comment by maximih** [comment]
I [discussed](https://botbot.me/mozilla/rust/2015-04-03/?msg=35748863&page=21) this one with @Kimundi on IRC some days ago and he said he has a fix for it in the pipeline

**Comment by Kimundi** [comment]
See #23952

**Comment by oli-obk** [comment]
so this can be closed now that #23952 is merged?

**Comment by steveklabnik** [comment]
The test case still fails with the same error.

**Comment by alexcrichton** [comment]
@steveklabnik I think last night's nightly didn't succeed, but I do believe @oli-obk is right in that @Kimundi fixed this in #23952. As a result, closing!
